release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md

```diff
@@ -68,18 +68,400 @@ A few backward-incompatible changes have been introduced during the Swan Lake Be
 
 #### New features
 
+##### Support for resource methods in client objects
+
+Client objects can now contain `resource` methods. These `resource` methods can only be accessed by client resource access actions. With this feature, `resource` methods are also allowed in object type descriptors.
+
+```ballerina
+import ballerina/io;
+
+type ClientObjectType client object {
+    resource function get greeting/[string name]() returns string;
+};
+
+ClientObjectType clientObj = client object {
+    resource function get greeting/[string name]() returns string {
+        return "Hello, " + name;
+    }
+};
+
+public function main() {
+    string result = clientObj->/greeting/James;
+    // Will print `Hello, James`
+    io:println(result);
+}
+```
+
+In the above example, the `greeting/[string name]` resource in `clientObj` is accessed with the `->/` call syntax. This notation signifies that it is a resource access action on a client object.
+
+The `greeting/James` resource access path segments after `->/` specify the target resource to access on the `clientObj`. The default resource method name will be`get` if it is not specified in the client resource access action.
+
+##### Support for creating maps with query expressions
+
+Maps can now be constructed using query expressions. A query expression that constructs a map must start with the `map` keyword. Further, the type of the `select` expression must be a tuple `[string, T]`. Then the query expression will create a value of type `map<T>`. Duplicate keys will be handled similar to a query expression constructing a table.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+	[string, int][] arr = [["A", 0], ["B", 1], ["C", 2], ["D", 3], ["A", 4]];
+
+	map<int>|error mapA = map from var element in arr
+                        	select [element[0], element[1]];
+	io:println(mapA); // {"A":4,"B":1,"C":2,"D":3}
+
+	map<int>|error mapB = map from var element in arr
+                        	select [element[0], element[1]]
+                        	on conflict error("Duplicate Key");
+	io:println(mapB); // error("Duplicate Key")
```

Comment:
```suggestion
    [string, int][] arr = [["A", 0], ["B", 1], ["C", 2], ["D", 3], ["A", 4]];

    map<int>|error mapA = map from var element in arr
        select [element[0], element[1]];
    io:println(mapA); // {"A":4,"B":1,"C":2,"D":3}

    map<int>|error mapB = map from var element in arr
        select [element[0], element[1]] on conflict error("Duplicate Key");
    io:println(mapB); // error("Duplicate Key")
```

Reply:
Will change to 

```suggestion
    [string, int][] arr = [["A", 0], ["B", 1], ["C", 2], ["D", 3], ["A", 4]];

    map<int>|error mapA = map from var element in arr
        					select [element[0], element[1]];
    io:println(mapA); // {"A":4,"B":1,"C":2,"D":3}

    map<int>|error mapB = map from var element in arr
							select [element[0], element[1]]
							on conflict error("Duplicate Key");
    io:println(mapB); // error("Duplicate Key")
```

to keep the rest of the query expression indented a bit more.